app/lib/ability.rb

```diff
@@ -58,8 +67,32 @@ def initialize(user)
     can :read, Organization, id: accessible_groups.pluck(:id)
     can :read, Site, id: accessible_groups.pluck(:id)
 
-    # Anyone can manage clients and client data in the groups they can access
-    can :manage, Client, vita_partner: accessible_groups
+    # This was overly permissive. We should work out what the permissions should
+    # be for each role and reduce this check. As we need to modify this, please
+    # break out the role and specify permissions more granularly
+    client_role_whitelist = [
+      :client_success, :admin, :org_lead, :site_coordinator,
+      :coalition_lead, :state_file_admin, :team_member
+    ].freeze
```

Comment:
These are every role I could find in the codebase to restore existing functionality. Moving from a 'everyone has permissions unless taken away' to a 'no one has permissions unless given' is secure by default while creating a little more friction. Moving permissions around might be painful in the short term but in the long term will save us from having to put role-specific things in views or controllers and will overall make the application easier to reason about.

With as many roles as we have, it may even make sense to break out into sub ability files for each role down the line.

Reply:
Great work. I totally agree abilities should almost always be whitelists.